src/data/sum.lean

```diff
@@ -111,12 +112,37 @@ funext $ λ x, sum.cases_on x (λ _, rfl) (λ _, rfl)
   sum.elim (f ∘ inl) (f ∘ inr) = f :=
 funext $ λ x, sum.cases_on x (λ _, rfl) (λ _, rfl)
 
-lemma elim_injective {α β γ : Sort*} {f : α → γ} {g : β → γ}
-  (hf : function.injective f) (hg : function.injective g)
- (hfg : ∀ a b, f a ≠ g b) : function.injective (sum.elim f g) :=
-λ x y, sum.rec_on x
-  (sum.rec_on y (λ x y hxy, by rw hf hxy) (λ x y hxy, false.elim $ hfg _ _ hxy))
-  (sum.rec_on y (λ x y hxy, false.elim $ hfg x y hxy.symm) (λ x y hxy, by rw hg hxy))
+open function (update update_eq_iff)
+
+@[simp] lemma update_elim_inl {α β γ} [decidable_eq α] [decidable_eq (α ⊕ β)]
+  {f : α → γ} {g : β → γ} {i : α} {x : γ} :
+  update (sum.elim f g) (inl i) x = sum.elim (update f i x) g :=
+update_eq_iff.2 ⟨by simp, by simp { contextual := tt }⟩
+
+@[simp] lemma update_elim_inr {α β γ} [decidable_eq β] [decidable_eq (α ⊕ β)]
+  {f : α → γ} {g : β → γ} {i : β} {x : γ} :
+  update (sum.elim f g) (inr i) x = sum.elim f (update g i x) :=
+update_eq_iff.2 ⟨by simp, by simp { contextual := tt }⟩
+
+@[simp] lemma update_inl_comp_inl {α β γ} [decidable_eq α] [decidable_eq (α ⊕ β)]
+  {f : α ⊕ β → γ} {i : α} {x : γ} :
+  update f (inl i) x ∘ inl = update (f ∘ inl) i x :=
+by conv { congr, rw [← elim_comp_inl_inr f, update_elim_inl, elim_comp_inl] }
+
+@[simp] lemma update_inl_comp_inr {α β γ} [decidable_eq α] [decidable_eq (α ⊕ β)]
+  {f : α ⊕ β → γ} {i : α} {x : γ} :
+  update f (inl i) x ∘ inr = f ∘ inr :=
+by conv { congr, rw [← elim_comp_inl_inr f, update_elim_inl, elim_comp_inr] }
+
+@[simp] lemma update_inr_comp_inl {α β γ} [decidable_eq β] [decidable_eq (α ⊕ β)]
+  {f : α ⊕ β → γ} {i : β} {x : γ} :
+  update f (inr i) x ∘ inl = f ∘ inl :=
+by conv { congr, rw [← elim_comp_inl_inr f, update_elim_inr, elim_comp_inl] }
+
+@[simp] lemma update_inr_comp_inr {α β γ} [decidable_eq β] [decidable_eq (α ⊕ β)]
+  {f : α ⊕ β → γ} {i : β} {x : γ} :
+  update f (inr i) x ∘ inr = update (f ∘ inr) i x :=
+by conv { congr, rw [← elim_comp_inl_inr f, update_elim_inr, elim_comp_inr] }
```

Comment:
I think this golfs to something like
```suggestion
function.update_comp_eq_of_injective f sum.inr_injective _ _
```
(using the lemma from data.set.function)